tracer/src/Datadog.Trace/Configuration/ImmutableIntegrationSettingsCollection.cs

```diff
@@ -15,64 +13,52 @@ namespace Datadog.Trace.Configuration
         /// <param name="disabledIntegrationNames">Additional integrations that should be disabled</param>
-        public ImmutableIntegrationSettingsCollection(
+        internal ImmutableIntegrationSettingsCollection(
             IntegrationSettingsCollection settings,
             HashSet<string> disabledIntegrationNames)
         {
-            _source = settings.Source;
-            _disabledIntegrations = disabledIntegrationNames ?? throw new ArgumentNullException(nameof(disabledIntegrationNames));
-            _settingsById = GetIntegrationSettingsById(settings, disabledIntegrationNames);
-            _settingsByName = GetIntegrationSettingsByName(settings, disabledIntegrationNames);
-            _valueFactory = name =>
-            {
-                if (IntegrationRegistry.Ids.TryGetValue(name, out var id))
-                {
-                    return _settingsById[id];
-                }
-
-                // We have no id for this integration, it will only be available in _settingsByName
-                var initialSettings = new IntegrationSettings(name, _source);
-                var isExplicitlyDisabled = _disabledIntegrations.Contains(name);
-
-                return new ImmutableIntegrationSettings(initialSettings, isExplicitlyDisabled);
-            };
+            Settings = GetIntegrationSettingsById(settings, disabledIntegrationNames);
         }
 
-        internal ICollection<string> DisabledIntegrations => _disabledIntegrations;
+        internal ImmutableIntegrationSettings[] Settings { get; }
 
         /// <summary>
-        /// Gets the <see cref="IntegrationSettings"/> for the specified integration.
+        /// Gets the <see cref="ImmutableIntegrationSettings"/> for the specified integration.
         /// </summary>
         /// <param name="integrationName">The name of the integration.</param>
         /// <returns>The integration-specific settings for the specified integration.</returns>
-        public ImmutableIntegrationSettings this[string integrationName] => this[new IntegrationInfo(integrationName)];
-
-        internal ImmutableIntegrationSettings this[IntegrationInfo integration]
+        public ImmutableIntegrationSettings this[string integrationName]
         {
             get
             {
-                return integration.Name == null
-                           ? _settingsById[integration.Id]
-                           : _settingsByName.GetOrAdd(integration.Name, _valueFactory);
+                if (IntegrationRegistry.TryGetIntegrationId(integrationName, out var integrationId))
+                {
+                    return Settings[(int)integrationId];
+                }
+
+                Log.Warning(
+                    "Accessed integration settings for unknown integration {IntegrationName}. Returning default settings",
+                    integrationName);
+
+                return new ImmutableIntegrationSettings(integrationName);
             }
         }
```

Comment:
Do we still need to keep this api? Probably best to I guess, but we shouldn't use it internally